lib/redmine_anonymous_watchers/watchers_controller_patch.rb

```diff
@@ -76,12 +76,19 @@ def create_with_anonymous
           mails = watcher[:mails].split(/[\s,]+/) || [watcher[:mail]]
           user_ids = []
           mails.each do |mail|
-            if(u = User.find_by_mail(mail))
-              user_ids << u.id
-            else
+            add_anonymous = false
+
+            u = User.find_by_mail(mail)
+            if !u || (u.is_a?(User) && u.groups.any? { |group| group.name == 'Anonymous Watchers' })
+              add_anonymous = true
+            end
+
+            if add_anonymous
               @watchables.each do |watchable|
                 AnonymousWatcher.create(:watchable => watchable, :mail => mail) if mail.present?
                end
+            else
+              user_ids << u.id
             end
           end
```

Comment:
Just squeeze this together some?

```suggestion
          mails.each do |mail|
            u = User.find_by_mail(mail)
            if !u || (u.is_a?(User) && u.groups.any? { |group| group.name == 'Anonymous Watchers' })
              @watchables.each do |watchable|
                AnonymousWatcher.create(:watchable => watchable, :mail => mail) if mail.present?
              end
            else
              user_ids << u.id
            end
          end
```

Reply:
... although we may want to discuss it because the syntax is mostly Greek to me.